File: assets/www/ios/plugins/ReadItLaterPlugin.js
//
//  ReadItLaterPlugin.js
//  ReadItLaterPlugin
//
//  Created by Tommy-Carlos Williams (tommy@devgeeks.org) on 16/03/12.
//  Copyright (c) 2012 Tommy-Carlos Williams. All rights reserved.
//  MIT Licensed
//

ReadItLaterPlugin = function() {
	
}

ReadItLaterPlugin.prototype.saveToReadItLater = function(successCallback, options) {
	// successCallback required
	if (typeof successCallback != "function") {
        console.log("ReadItLaterPlugin Error: successCallback is not a function");
        return;
    }
		
	PhoneGap.exec(successCallback, null, "ReadItLaterPlugin","saveToReadItLater",[options]);
};

PhoneGap.addConstructor(function(){
	if(!window.plugins) {
		window.plugins = {};
	}
	window.plugins.readItLaterPlugin = new ReadItLaterPlugin();
});

File: Wikipedia-iOS/Plugins/ReadItLater/ReadItLaterPlugin.m
//
//  ReadItLaterPlugin.m
//  ReadItLaterPlugin
//
//  Created by Tommy-Carlos Williams (tommy@devgeeks.org) on 16/03/12.
//  Copyright (c) 2012 Tommy-Carlos Williams. All rights reserved.
//  MIT Licensed
//

#import "ReadItLaterPlugin.h"
#import "ReadItLaterFull.h"

@implementation ReadItLaterPlugin

@synthesize callbackId;

-(PGPlugin*) initWithWebView:(UIWebView*)theWebView
{
    self = (ReadItLaterPlugin*)[super initWithWebView:theWebView];
    return self;
}

- (void)saveToReadItLater:(NSMutableArray*)arguments withDict:(NSMutableDictionary*)options
{
	self.callbackId = arguments.pop;
		
	NSString* RILURL = [options objectForKey:@"url"];
	NSString* RILTitle = [options objectForKey:@"title"];
	
	[ReadItLaterFull save:[NSURL URLWithString:RILURL] title:RILTitle];
	
	PluginResult* result = [PluginResult resultWithStatus: PGCommandStatus_OK messageAsInt: 1];
	[self.webView stringByEvaluatingJavaScriptFromString:[result toSuccessCallbackString: self.callbackId]];

}

- (void)dealloc
{	
	[callbackId release];
    [super dealloc];
}


@end
